[All Hosts](Security) ITP and third-party cookies (#2553)
OfficeDev/office-js-docs-pr: docs/develop/itp-and-third-party-cookies.md

---
title: Develop your Office Add-in to work with ITP when using third-party cookies
description: How to work with ITP and Office Add-ins when using third-party cookies
ms.date: 03/12/2021
localization_priority: Normal
---

# Develop your Office Add-in to work with ITP when using third-party cookies

If your Office Add-in requires third-party cookies, those cookies are blocked if Intelligent Tracking Prevention (ITP) is used by the browser runtime that loaded your add-in. You may be using third-party cookies to authenticate users, or for other scenarios, such as storing settings.

> If your Office Add-in and website must rely on third-party cookies, use the following steps to work with ITP.

Reviewer: ```suggestion
If your Office Add-in and website must rely on third-party cookies, use the following steps to work with ITP:
```

> 1. Set up [OAuth 2.0 Authorization](https://tools.ietf.org/html/rfc6749) with which the authenticating domain (in your case, the third-party that expects cookies) forwards an authorization token to your website which you consume and use to establish a first-party login session with a server-set Secure and [HttpOnly cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).

Reviewer: ```suggestion
1. Set up [OAuth 2.0 Authorization](https://tools.ietf.org/html/rfc6749) so that the authenticating domain (in your case, the third-party that expects cookies) forwards an authorization token to your website. Use the token to establish a first-party login session with a server-set Secure and [HttpOnly cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).
```

> 2. Use the [Storage Access API](https://webkit.org/blog/8124/introducing-storage-access-api/) with which the third-party can request permission to get access to its first-party cookies.

Reviewer: ```suggestion
2. Use the [Storage Access API](https://webkit.org/blog/8124/introducing-storage-access-api/) so that the third-party can request permission to get access to its first-party cookies.
```

    Current versions of Office on Mac and Office on the web both support this API.

>     If you are using cookies for purposes other than authentication, then you might consider using `localStorage` for your scenario.

Reviewer: ```suggestion
    If you're using cookies for purposes other than authentication, then consider using `localStorage` for your scenario.
```

> The following code sample shows how to use the Storage Access API.

Reviewer: ```suggestion
The following code sample shows how to use the Storage Access API:
```
